Library/Homebrew/cmd/pull.rb

```diff
@@ -14,15 +14,21 @@
 #   --bump:      For one-formula PRs, automatically reword commit message to our preferred format
 #   --clean:     Do not rewrite or otherwise modify the commits found in the pulled PR
 #   --ignore-whitespace: Silently ignore whitespace discrepancies when applying diffs
-#   --install:   Install changed formulae locally after pulling the patch
+#   --install:   Install or upgrade changed formulae locally after pulling each patch
 #   --resolve:   When a patch fails to apply, leave in progress and allow user to
 #                 resolve, instead of aborting
 #   --branch-okay: Do not warn if pulling to a branch besides master (useful for testing)
 
+# Developer notes: Because the `brew pull` actions may change the class definitions for
+# formulae and `brew` itself, it shells out to separate `brew` commands to get formula info
+# or do other actions. Each patch application may invalidate class definitions.
+
 require "utils"
 require "utils/json"
 require "formula"
 require "tap"
+require "net/http"
+require "net/https"
```

Comment:
I would prefer to shellout using `curl` then `net/https` to reduce code complexity.